onnxruntime/core/session/plugin_ep/ep_plugin_provider_interfaces.cc

```diff
@@ -207,27 +261,34 @@ PluginExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_vie
           this->Type(), this->Type(), /*node_unit_map*/ nullptr,
           node_grouping.fusion_options.drop_constant_initializers);
 
+      // Check if utils::CreateSupportedPartitions returned zero results.
+      // Happens if nodes have already been assigned to another EP.
+      if (capabilities.empty()) {
+        log_unsupported_node_info(node_grouping.nodes);
+        continue;
+      }
+
       if (capabilities.size() > 1) {
-        LOGS_DEFAULT(ERROR) << "OrtEp::GetCapability() set nodes that cannot be fused together. "
-                            << "Please ensure that the nodes provided to EpGraphSupportInfo_AddFusedNodes() do not "
+        LOGS(logger, ERROR) << "OrtEp::GetCapability() for " << Type() << " set nodes that cannot be fused together. "
+                            << "Please ensure that the nodes provided to EpGraphSupportInfo_AddNodesToFuse() do not "
                             << "have an unsupported node in any path between two of the supported nodes.";
```

Comment:
I think the `capabilities.size() > 1` case could also happen with a node that is assigned to another EP. e.g., if such a node is in between two supported nodes.

a few options I can think - treat that as an error, ignore a node_grouping like that, or add a capability (or capabilities) comprised of nodes that aren't assigned to another EP. not sure which is best.

it might be more consistent to warn and then ignore the entire node_grouping if it contains any nodes assigned to another EPs in all cases (single node, group with node assigned to another EP that partitions the group, and group with node assigned to another EP that doesn't).

Reply:
As you suggested, I'd prefer to log a warning and skip any node grouping that has any node assigned to another EP. I went ahead and just checked that once at the beginning of the for loop.

Then, the error checking for the other cases is simplified a bit: if we encounter something that could be considered a hard error, we log it and return immediately.

Let me know what you think.

Reply:
cool, that looks good to me. nice that it simplified the error handling too.